### PR TITLE
Improve select-datastore with cluster based filtering

### PR DESCRIPTION
**Pull request by shilovmaksim**
# Description
Updating select-datastore script with cluster name based sorting.

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images.CI/macos/select-datastore.ps1 [hunk 1]
```diff
@@ -57,8 +59,9 @@ function Select-DataStore {
     # 3. Choose a datastore with the minimal VM count < 2
 
     Write-Host "Start Datastore selection process..."
-    $allDatastores = Get-Datastore -Name $templateDatastore | Where-Object { $_.State -eq "Available" }
-    $availableDatastores = $allDatastores `
+    $clusterHosts = Get-VMHost | Where-Object {(Get-cluster -VMHost $_) -like $Cluster }
```
Comment (al-cheb) [inline images.CI/macos/select-datastore.ps1:62]: `Get-Cluster -Name $Cluster | Get-VMHost`

File: images.CI/macos/select-datastore.ps1 [hunk 2]
```diff
@@ -57,8 +59,9 @@ function Select-DataStore {
...
+    $availableClusterDatastores = $ClusterHosts | ForEach-Object {Get-Datastore -Name "*$_" | Where-Object -Property State -eq "Available"}
```
Comment (al-cheb) [inline images.CI/macos/select-datastore.ps1:63]: `$clusterHosts | Get-Datastore -Name $TemplateDatastore | Where-Object -Property State -eq "Available"`

File: images.CI/macos/select-datastore.ps1 [hunk 3]
```diff
@@ -36,7 +36,9 @@ param(
     [ValidateNotNullOrEmpty()]
     [string]$VIPassword,
 
-    [string]$TagCategory = "Busy"
+    [string]$TagCategory = "Busy",
+
+    [string]$Cluster
```
Comment (al-cheb) [inline images.CI/macos/select-datastore.ps1:41]: Add and pass $cluster parameter to Select-DataStore function

File: images.CI/macos/select-datastore.ps1 (hunk 1 above)
Comment (shilovmaksim) [reply images.CI/macos/select-datastore.ps1]: Updated

File: images.CI/macos/select-datastore.ps1 (hunk 2 above)
Comment (shilovmaksim) [reply images.CI/macos/select-datastore.ps1]: Updated

File: images.CI/macos/select-datastore.ps1 (hunk 3 above)
Comment (shilovmaksim) [reply images.CI/macos/select-datastore.ps1:41]: Updated

**Review by al-cheb — DISMISSED** [review]

**Review by miketimofeev — DISMISSED** [review]

**Review by al-cheb — DISMISSED** [review]

**Review by miketimofeev — APPROVED** [review]

**Review by al-cheb — DISMISSED** [review]

**Review by miketimofeev — APPROVED** [review]

**Review by maldito187 — COMMENTED** [review]
@maldito-dot

**Review by dmitry-shibanov — APPROVED** [review]

**Pull request merged**